frontend-runitup/src/utils/mapUtils.js

```diff
@@ -200,11 +260,36 @@ export const addStartMarker = (map, coordinates, locationName) => {
 };
 
 // Function to clear the route from the map
+let elevationChart = null;
+let elevationMarkers = [];
+
 export const clearRoute = (map) => {
+  // Remove route-related layers
+  if (map.getLayer("route-line")) {
+    map.removeLayer("route-line");
+  }
+
+  // Remove any other layers that might be using the 'route' source
+  map.getStyle().layers.forEach((layer) => {
+    if (layer.source === "route") {
+      map.removeLayer(layer.id);
+    }
+  });
+
+  // Remove the route source
   if (map.getSource("route")) {
-    map.removeLayer("route");
     map.removeSource("route");
   }
+
+  // Clear elevation markers
+  elevationMarkers.forEach((marker) => marker.remove());
+  elevationMarkers = [];
+
+  // Clear existing chart data
+  if (elevationChart) {
+    elevationChart.data.datasets[0].data = [];
+    elevationChart.update();
+  }
```

Comment:
This could be its own function to like "clear<xxxx>" since you're clearing elevation markers and existing chart data